lua/barbar.lua

```diff
@@ -74,6 +74,45 @@ function barbar.setup(options)
     }
   )
 
+  create_user_command(
+    'BufferGotoPinned',
+    function(tbl)
+      local index = tonumber(tbl.args)
+      if not index then
+        return notify(
+          'Invalid argument to ' .. markdown_inline_code':BufferGotoPinned',
+          vim.log.levels.ERROR
+        )
+      end
+
+      api.goto_buffer_pinned(index)
+    end,
+    {
+      desc = 'Go to the buffer at the specified index',
+      nargs = 1,
+    }
```

Comment:
Do you think it's worth extracting `:BufferGoto`'s `complete` function so that these new functions can provide completions as well?

Reply:
I don't think completion is really useful for this command :| Either you know the bufnr you want, or you don't. See which bufnrs are available won't help you make a decision, unless the completion also somehow indicates the buffer name/path.